examples/flow-counter/containers/App.js

```diff
@@ -0,0 +1,36 @@
+/* @flow */
+
+import { connect } from 'react-redux';
+import Counter from '../components/Counter';
+import type { CounterProps } from '../components/Counter';
+import { increment, decrement, incrementIfOdd, incrementAsync } from '../actions/counter';
+import { AppState, Dispatchable } from '../types';
+import type { Dispatch } from 'redux';
+import type { Component } from 'react';
+
+type StateProps = AppState;
+
+function mapStateToProps(state : AppState) : StateProps {
+  return {
+    counter: state.counter
+  };
+}
+
+type DispatchProps = {
+  increment: () => any,
+  decrement: () => any,
+  incrementIfOdd: () => any,
+  incrementAsync: () => any
+}
+
+function mapDispatchToProps(dispatch: Dispatch<Dispatchable>) : DispatchProps {
+  return {
+    increment: (...args) => dispatch(increment(...args)),
+    decrement: (...args) => dispatch(decrement(...args)),
+    incrementIfOdd: (...args) => dispatch(incrementIfOdd(...args)),
+    incrementAsync: (...args) => dispatch(incrementAsync(...args)),
+  };
+}
+// TODO: Cannot make this to correctly type check that the resulting props match
+// the Counter prop types
+export default connect(mapStateToProps, mapDispatchToProps)(Counter);
```

Comment:
I still need to make this type check correctly, I cannot find a way to specify in the type signature that the resulting props should be the merge of `StateProps` and `DispatchProps` from above (both should result in `CounterProps` that is the props type for the `Counter` component, but for some reason if I change any of the types from here, it still passes with no errors.